modules/Bio/EnsEMBL/Compara/RunnableDB/HomologyAnnotation/ParsePAFforBHs.pm

```diff
@@ -60,6 +60,7 @@ sub write_output {
     my $query_mem_adap = $self->compara_dba->get_SeqMemberAdaptor;
 
     # MLSS will not exist for hit_gdb: it is a reference and belongs to a different db
+    my $ss   = $self->_create_species_set($self->compara_dba->get_GenomeDBAdaptor->fetch_by_dbID($query_gdb_id), $ref_db->get_GenomeDBAdaptor->fetch_by_dbID($hit_gdb_id));
```

Comment:
I have just found out that `$mlss_adap->store()` will automatically store the `method_link` and `species_set` if any of these doesn't exist in the database. Anyway, I think you can still leave everything as one method (like you had before), since most of the code was already in `_create_superficial_mlss()` and that way you only fetch twice via `GenomeDBAdaptor` instead of four times?

Reply:
The tricky thing with this was that the `species_set` was being created multiple times simultaneously in `_create_superficial_mlss()` (when the same mlss was being run in jobs in parallel)- this is why I changed it in this way